src/api/python/hidra/utils/utils_general.py

```diff
@@ -371,6 +372,35 @@ def create_sub_dirs(dir_path, subdirs, dirs_not_to_create=()):
         raise OSError
 
 
+class TempFile:
+    def __init__(self, temp_file, target_path):
+        self.temp_file = temp_file
+        self.target_path = str(target_path)
+
+    def seek(self, offset, whence=0):
+        return self.temp_file.seek(offset, whence)
+
+    def truncate(self, size=None):
+        return self.temp_file.truncate(size)
+
+    def write(self, data):
+        return self.temp_file.write(data)
+
+    def close(self):
+        if not self.temp_file.closed:
+            self.temp_file.close()
+            os.chmod(self.temp_file.name, 0o644)
+            os.rename(self.temp_file.name, self.target_path)
+
+
+def open_tempfile(filepath):
+    filepath = Path(filepath)
```

Comment:
Similar to @criticanalytic's comment in the other PR, changing a `string` object to a `Path` object feels surprising (again, from my static types world view). A new variable name  would dispel confusion, and is, I assume, not much more expensive than reusing the old variable.